core/src/main/java/jenkins/util/CustomUrlValidator.java

```diff
@@ -56,44 +56,37 @@
      * <li> custom TLD like "intern" that is not standard but could be registered locally in a network</li>
      * </ol>
      */
-    private static String DOMAIN_REGEX = System.getProperty(
-            UrlHelper.class.getName() + ".DOMAIN_REGEX", 
-            "^" + 
-            "\\w" + // must start with letter / number / underscore
-                "(-*(\\.|\\w))*" +// dashes are allowed but not as last character
-                "\\.*" + // can end with zero (most common), one or multiple dots 
-                "(:\\d{1,5})?" + // and potentially the port specification
-                "$"
+    private static final String DOMAIN_REGEX = System.getProperty(
+            CustomUrlValidator.class.getName() + ".DOMAIN_REGEX",
```

Comment:
Thanks for the feedback. I mined the repo and found tight coupling as area for improvement and tried to make a little change to address this. My other thought would be to keep the URLHelper class and extract the CustomUrlValidator but due to the system property this would not achieve what I originally intended.